Implement shortest path on map_srv, upgrade rebar3

priestjim opened the pull request:
- Implemented find shortest path by weight using Dijkstra's algorith
  by exposing a shortest_path/2 function
- Upgraded rebar3
- Updated sys.config with more paths

Commit:
Implement shortest path on map_srv, upgrade rebar3

- Implemented find shortest path by weight using Dijkstra's algorith
  by exposing a shortest_path/2 function
- Upgraded rebar3
- Updated sys.config with more paths

blt commented:
I think the rebar3 upgrade would be better as a separate PR to evaluate there. Just because it's potentially something that'll introduce new behaviours.

blt commented on apps/fahrradboten/src/dijkstra.erl:
```diff
@@ -0,0 +1,54 @@
+% Implementation copied from http://rosettacode.org/wiki/Dijkstra%27s_algorithm#Erlang
+
+-module(dijkstra).
+-include_lib("eunit/include/eunit.hrl").
```
This line can be removed. It's already held in the TEST block.

blt commented on apps/fahrradboten/src/dijkstra.erl:
```diff
@@ -0,0 +1,54 @@
+    },
+    {Cost, Path}   = dijkstrafy(Graph, a, e),
+    {20,[a,c,f,e]} = {Cost, Path},
+    io:format(user, "The total cost was ~p and the path was: ", [Cost]),
```
These should be assertions. You an use the `?assertMatch` macro for this.

blt commented:
This is great! I would suggest to swap out `map:path/2` with your new `shortest_path` implementation. It's an improvement that doesn't break the contract with existing clients.

Commit:
Add dijkstra conversion and shortest path test

Commit:
Switch to shortest path, apply PR comments

- Switch to to Dijkstra-based shortest path instead of the digraph one
- Reverted rebar3 to the original version
- Removed extra eunit header
- Updated dijkstra test with assertions
